repo_policy_compliance/blueprint.py

```diff
@@ -212,6 +215,26 @@ def workflow_dispatch_check_run(body: WorkflowDispatchInput) -> Response:
     return Response(status=204)
 
 
+@repo_policy_compliance.route(PUSH_CHECK_RUN_ENDPOINT, methods=["POST"])
+@auth.login_required(role=RUNNER_ROLE)
+@validate()
+def push_check_run(body: PushInput) -> Response:
+    """Check whether a push run should proceed.
+
+    Args:
+        body: The request body after it is validated.
+
+    Returns:
+        Either to proceed with the run or an error not to proceed with a reason why.
+    """
+    policy_document = _get_policy_document()
+
+    if (report := push(input_=body, policy_document=policy_document)).result == Result.FAIL:
+        return Response(response=report.reason, status=403)
```

Comment:
Minor things, but I'd use for http.HTTPStatus instead of the codes

Reply:
thats a good one - didn't know about that module. Should be fixed on next push